image-loader/app/ImageLoaderComponents.scala

```diff
@@ -18,15 +18,20 @@ class ImageLoaderComponents(context: Context) extends GridComponents(context, ne
 
   val store = new ImageLoaderStore(config)
   val imageOperations = new ImageOperations(context.environment.rootPath.getAbsolutePath)
-
   val notifications = new Notifications(config)
   val downloader = new Downloader()
   val uploader = new Uploader(store, config, imageOperations, notifications)
-
   val projector = Projector(config, imageOperations)
-
+  val checkQuarantineFlag = config.uploadToQuarantineEnabled 
+  val quarantineUploader: Option[QuarantineUploader] = checkQuarantineFlag match {
+    case true =>{
+      val quarantineStore = new QuarantineStore(config)
+      Some(new QuarantineUploader(quarantineStore, config))
+    }
+    case false => None
+  }
```

Comment:
If you make quarantineBucket optional then you can do something like this to deal with the Boolean and Option[String]:

```suggestion
  val quarantineUploader: Option[QuarantineUploader] = (config.uploadToQuarantineEnabled, config.quarantineBucket) match {
    case (true, Some(bucketName)) =>{
      val quarantineStore = new QuarantineStore(bucketName, config)
      Some(new QuarantineUploader(quarantineStore, config))
    }
    case (true, None) => throw new IllegalArgumentException("Quarantining is enabled but no bucket is configured")
    case (false, _) => None
  }
```

The advantage of this approach is that a bad config (when quarantine is enabled but no bucket is specified) fails fast.

Reply:
Note that this now passes the quaratine bucket directly into QuarantineUploader (which deals with the config being an option).